### PR TITLE
KAN-156: Tutorial and example to analyse ros bags from boston dynamics

### PR DESCRIPTION
**Pull request by AnthonyCvn**

### CODE REVIEW

File: blog/2025-04-30-boston-dynamic-example/index.mdx [hunk 1]
```diff
@@ -0,0 +1,266 @@
+---
+title: "How to work with ROS bag files and classify robot movements using Python"
+description: "Learn how to process .bag files and train models to classify different robot movements"
+authors: ekaterina
+tags: [robotics, ROS, tutorials]
+slug: boston-dynamic-example
+date: 2025-04-30
+# image: ./img/demo_setup.png
```
Comment (AnthonyCvn) [inline blog/2025-04-30-boston-dynamic-example/index.mdx:8]: an image is needed here for social media banners

File: blog/2025-04-30-boston-dynamic-example/index.mdx [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+---
+
```
Comment (AnthonyCvn) [inline blog/2025-04-30-boston-dynamic-example/index.mdx:10]: here you can add the same image as the one you create for social media as an introduction image when you are browsing the page "reduct.store/blog"

File: blog/2025-04-30-boston-dynamic-example/index.mdx [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+In this tutorial, we’ll show you how to:
+
+- extract motion data from `.bag` files
+- create basic velocity features
+- train a classification model to recognize different types of robot movements
+
+We'll use the `bagpy` library to process `.bag` files and basic machine learning techniques for classification.
+
+Although the examples use a Boston Dynamics Spot robot (moving forward, sideways, and rotating), you can adapt the code for your own recordings. The original data can be found [here](http://ptak.felk.cvut.cz/darpa-subt/qualification_videos/spot/).
```
Comment (AnthonyCvn) [inline blog/2025-04-30-boston-dynamic-example/index.mdx:21]: the link text "here" should be descriptive (that's an SEO thing to do) and you can make the link in bold. For example:

The [**original data from Boston Dynamics Spot robot**](http://ptak.felk.cvut.cz/darpa-subt/qualification_videos/spot/)..

File: blog/2025-04-30-boston-dynamic-example/index.mdx [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+
+{/* truncate */}
+
+### **Install Required Libraries**
```
Comment (AnthonyCvn) [inline blog/2025-04-30-boston-dynamic-example/index.mdx:25]: you don't need to add bold after `###`

File: blog/2025-04-30-boston-dynamic-example/index.mdx [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+description: "Learn how to process .bag files and train models to classify different robot movements"
+authors: ekaterina
+tags: [robotics, ROS, tutorials]
+slug: boston-dynamic-example
+date: 2025-04-30
+# image: ./img/demo_setup.png
+---
+
+Working with real-world robot data often starts with `.bag` files — a standard format for recording ROS (Robot Operating System) messages. Whether you’re logging odometry, camera frames, LiDAR or IMU data, bag files contain valuable data on how a robot interacts with the world.
+
```
Comment (AnthonyCvn) [inline blog/2025-04-30-boston-dynamic-example/index.mdx:12]: you can also explain shortly that we saw different ways to store ROS data in the article [**3 Ways to Store ROS Topics**](https://www.reduct.store/blog/store-ros-topics) and here we assume that it's stored in rosbags and we want to analyse it and train an ML model in python.

File: blog/2025-04-30-boston-dynamic-example/index.mdx [hunk 6]
```diff
@@ -0,0 +1,266 @@
+---
+title: "How to work with ROS bag files and classify robot movements using Python"
```
Comment (AnthonyCvn) [inline blog/2025-04-30-boston-dynamic-example/index.mdx:2]: "classify robot movements" is the example used in the tutorial, but would make the title more about learning to work with ROS bag files and use sklearn to train an ML model.

File: blog/2025-04-30-boston-dynamic-example/index.mdx [hunk 7]
```diff
@@ -0,0 +1,266 @@
...
+description: "Learn how to process .bag files and train models to classify different robot movements"
```
Comment (AnthonyCvn) [inline blog/2025-04-30-boston-dynamic-example/index.mdx:3]: description is quite important for google. you can make it a bit longer. Like "In this tutorial you will lear... We will use ..."

File: blog/2025-04-30-boston-dynamic-example/index.mdx [hunk 8]
```diff
@@ -0,0 +1,266 @@
...
+With an F1 score of 0.976, this classifier performs well, but you can always experiment to optimize the model based on your own data.
+
+After evaluating the Random Forest model, we can look at **Feature Importance** to see velocity components contributed most to distinguishing the movement types. This is useful for Decision Tree and Random Forest models, which automatically rank features based on their importance.
+
+```python
+plot_feature_importance(best_model, X_columns)
+```
+
+![Feature Importance](./img/feature_importance.png)
+
```
Comment (AnthonyCvn) [inline blog/2025-04-30-boston-dynamic-example/index.mdx:256]: you can add here a small section "Best practices" and explain in a few bullet points what to be careful about. For example if ROS bag files is too big, you can split it in 1-5 minutes episodes; split topics with lightweight topics in on BAG and images or LiDAR in another one if any. Also best practices about the use of sklearn (for example to use randomize search instead of grid search if accuracy is not so good ; try different methods; consider ensemble methods or deep-learning models; avoid overfitting and split dataset into train/val/test)

**Review by AnthonyCvn — COMMENTED** [review]
@EkaterinaMarova the tutorial looks good overall. You can find my review in the comments for a few details.
